perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/ConsentsManagerMethod.java

```diff
@@ -81,5 +83,27 @@ public ConsentHub call(ApiCaller ac, Deserializer parms) throws PerunException {
 			return ac.getConsentsManager().updateConsentHub(ac.getSession(),
 				parms.read("consentHub", ConsentHub.class));
 		}
+	},
+
+	/*#
+	 * Returns consent with changed status.
+	 *
+	 * @param consent int Consent <code>id</code>
+	 * @param status String UNSIGNED | GRANTED | REVOKED
+	 * @param user int User <code>id</code>
+	 *
+	 * @throws ConsentNotExistsException
+	 * @throws PrivilegeException
+	 * @throws UserNotExistsException
+	 * @throws InvalidConsentStatus
+	 *
+	 * @return Consent
+	 */
+	changeConsentStatus {
+		@Override
+		public Consent call(ApiCaller ac, Deserializer params) throws PerunException {
+			return ac.getConsentsManager().changeConsentStatus(ac.getSession(), params.readInt("consent"), ConsentStatus.valueOf(params.readString("status")), params.readInt("user"));
```

Comment:
```suggestion
			return ac.getConsentsManager().changeConsentStatus(ac.getSession(), ac.getConsentById(parms.readInt("consent")), ConsentStatus.valueOf(params.readString("status")));
```
In RPC, we usually get the object by the ID and the object is then passed to the Entry layer instead of the ID. Then we won't have to get the object by ID in the entry layer. You will have to create a simple method `getConsentById` in ApiCaller.